### PR TITLE
fix(elasticsearch): fixes out of order runId writes

### PR DESCRIPTION
**Pull request by david-leifker**
fix(elasticsearch): fixes out of order writes of runId, only update runId after document creation
fix(usageclient): Add retry policy to usage client
refactor(elasticsearch): Clean up syntax for upserts

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-service/restli-client/src/main/java/com/linkedin/entity/client/EntityClient.java [hunk 1]
```diff
@@ -295,7 +295,7 @@ default String wrappedIngestProposal(@Nonnull MetadataChangeProposal metadataCha
       @Nonnull final Authentication authentication, final boolean async) {
     try {
       return ingestProposal(metadataChangeProposal, authentication, async);
-    } catch (RemoteInvocationException e) {
+    } catch (Exception e) {
```
Comment (RyanHolstien) [inline metadata-service/restli-client/src/main/java/com/linkedin/entity/client/EntityClient.java:298]: Revert loosened constraint if unintended

File: metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/elasticsearch/ElasticsearchConnector.java [hunk 2]
```diff
@@ -46,11 +47,12 @@ private static DeleteRequest createDeleteRequest(@Nonnull ElasticEvent event) {
...
   private UpdateRequest createUpsertRequest(@Nonnull ElasticEvent event) {
-    final IndexRequest indexRequest = new IndexRequest(event.getIndex()).id(event.getId()).source(event.buildJson());
-    return new UpdateRequest(event.getIndex(), event.getId()).doc(event.buildJson())
-        .detectNoop(false)
-        .retryOnConflict(_numRetries)
-        .upsert(indexRequest);
+    return new UpdateRequest(
+            event.getIndex(), event.getId())
+            .detectNoop(false)
+            .docAsUpsert(true)
```
Comment (RyanHolstien) [inline metadata-jobs/mae-consumer/src/main/java/com/linkedin/metadata/kafka/elasticsearch/ElasticsearchConnector.java:53]: Nice!

**Review by RyanHolstien — APPROVED** [review]

File: metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/update/ESWriteDAO.java [hunk 3]
```diff
@@ -62,7 +62,12 @@ public void deleteDocument(@Nonnull String entityName, @Nonnull String docId) {
    */
   public void applyScriptUpdate(@Nonnull String entityName, @Nonnull String docId, @Nonnull String script) {
     final String indexName = indexConvention.getIndexName(entityRegistry.getEntitySpec(entityName));
-    bulkProcessor.add(new UpdateRequest(indexName, docId).retryOnConflict(numRetries).script(new Script(script)));
+    UpdateRequest updateRequest = new UpdateRequest(indexName, docId)
+            .detectNoop(false)
+            .scriptedUpsert(true)
```
Comment (jjoyce0510) [inline metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/update/ESWriteDAO.java:67]: Thank you!!!

File: metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/AspectResource.java [hunk 4]
```diff
@@ -153,15 +155,18 @@ public Task<String> ingestProposal(
...
-        Urn urn = _entityService.ingestProposal(metadataChangeProposal, auditStamp, asyncBool).getUrn();
-        additionalChanges.forEach(proposal -> _entityService.ingestProposal(proposal, auditStamp, asyncBool));
-        tryIndexRunId(urn, metadataChangeProposal.getSystemMetadata(), _entitySearchService);
+        EntityService.IngestProposalResult result = _entityService.ingestProposal(metadataChangeProposal, auditStamp, asyncBool);
+        Urn urn = result.getUrn();
+
+        AspectUtils.getAdditionalChanges(metadataChangeProposal, _entityService)
+                .forEach(proposal -> _entityService.ingestProposal(proposal, auditStamp, asyncBool));
+
+        if (!result.isDidUpdate()) {
```
Comment (jjoyce0510) [inline metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/AspectResource.java:167]: Why is this check required prior to attempting to index the run id? We should always index the latest run ID so we can show which entities were touched by which run (in the Ingestion Run Summary)

**Pull request merged**